.github/workflows/build-containers.yml

```diff
@@ -75,120 +75,192 @@ jobs:
           image_json=$(echo -n "${images:-dummy}" | jq -Rcs '.|split("\n")')
           echo "json=$image_json" >> $GITHUB_OUTPUT
 
-  build:
+  check-config:
     runs-on: ubuntu-latest
-    needs: build-image-list
-    if: needs.build-image-list.outputs.images != '["dummy"]'
-    # Prevent a single build failure from killing the workflow.
-    # This is safe since subsequent pushes should fail on cache load
-    continue-on-error: true
+    needs: [make-date-tag, build-image-list]
+    outputs:
+      image_matrix: ${{ steps.create-matrix.outputs.image_matrix }}
+      default_image_matrix: ${{ steps.create-default-matrix.outputs.default_image_matrix }}
     strategy:
       fail-fast: false
       matrix:
         image: ${{ fromJson(needs.build-image-list.outputs.images) }}
-        yum_repo: ['development', 'testing', 'release']
-        series:
-          - name: '3.6'
-            os: 'el7'
-          - name: '23'
-            os: 'el9'
     steps:
-      - name: Build ${{ matrix.image }}:${{ matrix.series }}-${{ matrix.yum_repo }}
-        continue-on-error: ${{ matrix.yum_repo == 'development' }}
-        uses: opensciencegrid/build-container-action@v0.6.0
+      - uses: actions/checkout@v3
         with:
-          osg_series: ${{ matrix.series.name }}
-          osg_repo: ${{ matrix.yum_repo }}
-          context: ${{ matrix.image }}
-          base_os: ${{ matrix.series.os }}
-
-  push:
+          fetch-depth: 0
+  
+      - name: Check for build-config.json in ${{ matrix.image }}
+        id: check-jsonfile
+        run: |
+          if [ -f "${{ matrix.image }}/build-config.json" ]; then
+            echo "build-config.json exists in ${{ matrix.image }}"
+            echo "config_exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "No build-config.json found in ${{ matrix.image }}"
+            echo "config_exists=false" >> $GITHUB_OUTPUT
+          fi
```

Comment:
Instead of returning a boolean, let's change this step to return the path to the config file. If the custom config exists, use it, if not, use the default config